src/Shell32.Desktop/Shell32+SHITEMID.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+
+namespace PInvoke
+{
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SHITEMID"/> nested type.
+    /// </content>
+    public partial class Shell32
+    {
+        /// <summary>
+        /// Defines an item identifier.
+        /// </summary>
+        /// <remarks>Used by <see cref="ITEMIDLIST"/></remarks>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SHITEMID
+        {
+            /// <summary>
+            /// The size of identifier, in bytes, including cb itself.
+            /// </summary>
+            ushort cb;
+
+            /// <summary>
+            /// A variable-length item identifier.
+            /// </summary>
+            public byte[] abID;
```

Comment:
This is not a structure. abID is indeed a byte array of the identifier as it is described on [MSDN](https://msdn.microsoft.com/en-us/library/windows/desktop/bb759800(v=vs.85).aspx).

If you mean, to be able to make a ITEMIDLIST* on the function I will need to make a byte* here, ok.

I did a pointer on a struct on the User32 pull-request (MOUSEEVT) and had no success on using the IntPtr accessor. All my pull-requests are backed by in-house PInvokes that I had on the system that I am migrating to use this project. Unfortunately I could not solve the void* on that structure for now. I really disliked this practice. Sorry.